### DISCUSSION TITLE
Run from linux console.

### DISCUSSION THREAD

**Discussion by MaverickPOL** [shirokumacode/overwatch-omnic-rewards#5] [Ideas]
It is possible or possible to create version that can be run from linux console without GUI? So it can be run on Raspberry PI directly from linux console?

**Comment by shirokumacode** [comment]
~~You can run the current the app directly on console with argument _-q_ or _--quiet_ to suppress the system tray icon. I would recommend to also add the argument _-v_ or _--verbose_ to get at least feedback on what the app is doing. If you intend to run it this way, don't forget to copy/build the config.json previously, otherwise you won't be able to set an account without the GUI.~~
~~Running with executable:~~
```shell
./omnic_rewards_Linux -q -v
```
~~Running from repository directly:~~
```shell
python gui.py -q -v
```

**_Edit:_**: --quiet argument was replaced by --cli -> python app.py --cli

Alternatively, If you are proficient in Python you can create/use the example in utils/example.py to create what you want, personalized. It's very simple to do, and you can even turn into a systemd service if you want. 

If you run into any issues, feel free to reply and I'll try to help as best I can.

**Comment by MaverickPOL** [comment]
Unfortunatelly i have absolutelly no skill in python :(

I try to run this as quiet and verbose but i only get this:

./omnic_rewards_Linux -q -v
qt.qpa.xcb: could not connect to display
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, webgl, xcb.

Maby becouse i try to run this on machine where is only linux console.

If you will have time, please consider make "command line version", it will be very helpfull for noob like me ;)

**Comment by shirokumacode** [comment]
That is a Qt error. It is out-of-scope of the app. The app should run fine on the Raspberry Pi, providing you have a X11 server running and have a system tray. This depends on the linux distro you have/and your configurations. And unfortunately I don't have a Raspberry Pi so I can try to reproduce this and fix any build issues if there are any. Also not sure if the Qt build packaged is completely functional with Raspberry Pi (this was packaged under Ubuntu), you might need to build the app yourself or run directly from the repository.

I am a complete beginner with Qt, but after doing a small search it seems that: 
- If you don't have a X11 server running, I might be able to create a version that doesn't need it
- If you do have one running and system tray, you might need to set the DISPLAY environment variable ($ export DISPLAY=:0.0) and then run the app

Anyway you can try build/run the app directly from the repository, so that we can find if there are any build issues. This is very simple to do. You need Python 3.9. Check the section of the README->How to Run

If I have time today I'll create a version that doesn't need system tray/gui entirely (possibly X11 also), to see if it addresses your use case.

**Reply by MaverickPOL** [reply]
I don't have X11, probably thats the reason.

If its not too big trouble, i will be happy if you can create version for command line only without x11.

**Comment by Yayady1999** [comment]
Open terminal
In the terminal window you should be able to see the following prompt:

pi@raspberrypi ~ $

This shows your username and the hostname of the Pi. Here the username is pi and the hostname is raspberrypi.
Navigating and browsing your Pi

One of the key aspects of using a terminal is being able to navigate your file system. Go ahead and type ls -la into the Terminal window, and then hit the RETURN key. 

the ls command lists the contents of the directory that you are currently in (your present working directory). The -la component of the command is what’s known as a 'flag'. Flags modify the command that’s being run. In this case the l displays the contents of the directory in a list, showing data such as their sizes and when they were last edited, and the a displays all files, including those beginning with a ., known as 'dotfiles'. Dotfiles usually act as configuration files for software and as they are written in text, they can be modified by simply editing them.

In order to navigate to other directories the change directory command, cd, can be used. You can specify the directory that you want to go to by either the 'absolute' or the 'relative' path. So if you wanted to navigate to the python_games directory, you could either do cd /home/pi/python_games or just cd python_games (if you are currently in /home/pi). There are some special cases that may be useful: ~ acts as an alias for your home directory, so ~/python_games is the same as /home/pi/python_games; . and .. are aliases for the current directory and the parent directory respectively, e.g. if you were in /home/pi/python_games, cd .. would take you to /home/pi.

**Reply by shirokumacode** [reply]
This is irrelevant to the discussion

**Comment by shirokumacode** [comment]
@MaverickPOL Hey sorry for only answering now, but I have been extremely busy this past week

I have created a new version that shouldn't need X11 or a system tray. It should be in the releases section (v1.2) when the upload is done.

It should work as I was able to reproduce your error with a ubuntu server instance under docker.

To run it in Command Line mode, please use the -c or --cli argument:
```shell
./omnic_rewards_Linux --cli
```
You should copy your config.json created previously or create your own (only accountid needs to be set).

Also the app should respond to signals SIGINT and SIGTERM, and this will be your method to exit/quit the app. Simply send any of these signals and the app should exit gracefully. Ctrl+C on the terminal should do the trick.

If you have any doubts/troubles/requests hit me up, I'll try to answer quickly.

**Comment by MaverickPOL** [comment]
./omnic_rewards_Linux --cli
INFO:settings:Settings loaded
INFO:__main__:CLI Mode enabled
INFO:checkviewer:Starting checkviewer
INFO:checkviewer:Checking if live
INFO:cli:Not Live - 5min until next check


Looks that it's working great.
Thank you so much for this cli version.  :)
